core/src/testFixtures/java/bio/terra/pearl/core/factory/StudyEnvironmentFactory.java

```diff
@@ -62,4 +69,26 @@ public StudyEnvironment buildPersisted(EnvironmentName envName, UUID studyId, St
         return studyEnvironmentService.create(studyEnv);
     }
 
+    public StudyEnvironmentBundle buildBundle(String testName, EnvironmentName envName) {
+        PortalEnvironment portalEnvironment = portalEnvironmentFactory.buildPersisted(testName);
+        Study study = studyFactory.buildPersisted(portalEnvironment.getPortalId(), testName);
+        StudyEnvironment studyEnvironment = buildPersisted(envName, study.getId(), testName);
+        Portal portal = portalService.find(portalEnvironment.getPortalId()).get();
+        return StudyEnvironmentBundle.builder()
+                .study(study)
+                .studyEnv(studyEnvironment)
+                .portal(portal)
+                .portalEnv(portalEnvironment)
+                .build();
+    }
```

Comment:
Nice, seems like a quick way to get everything you need. Not a concern yet, but re: longer execution times, we should keep that in mind and periodically check in on execution time. One of the many nice things about this project is that it's pretty quick to iterate on tests (running tests locally, waiting for CI to run tests, etc). Easier to deal with and remediate any slowness when we stay ahead of those things.

Reply:
yeah, agree on fast tests being nice.  my main goal is to make sure we're keeping our tests independent.  As long as they're independent, and don't rely on, e.g. populating a big shared DB, we can make them fast in CI by parallelizing them.